rest-service/manager_rest/rest/resources_v3_1/blueprints.py

```diff
@@ -116,6 +121,71 @@ def delete(self, blueprint_id, **kwargs):
             force=query_args.force)
         return None, 204
 
+    @authorize('blueprint_upload')
+    @rest_decorators.marshal_with(models.Blueprint)
+    def patch(self, blueprint_id, **kwargs):
+        """
+        Update a blueprint.
+
+        Used for updating the blueprint's state (and error) while uploading,
+        And updating the blueprint's other attributes upon a successful upload.
+        This method is for internal use only.
+        """
+        if not request.json:
+            raise IllegalActionError('Update a blueprint request must include'
+                                     ' at least one parameter to update')
+
+        request_schema = {
+            'plan': {'type': db.PickleType, 'optional': True},
+            'description': {'type': text_type, 'optional': True},
+            'main_file_name': {'type': text_type, 'optional': True},
+            'visibility': {'type': text_type, 'optional': True},
+            'state': {'type': text_type, 'optional': True},
+            'error': {'type': text_type, 'optional': True},
+        }
+        request_dict = rest_utils.get_json_and_verify_params(request_schema)
+
+        invalid_params = set(request_dict.keys()) - set(request_schema.keys())
+        if invalid_params:
+            raise BadParametersError(
+                "Unknown parameters: {}".format(','.join(invalid_params))
+            )
+        sm = get_storage_manager()
+        blueprint = sm.get(models.Blueprint, blueprint_id)
+
+        # set blueprint state
+        state = request_dict.get('state', None)
```

Comment:
`dict.get(<key>)` default behavior is to return `None` in case this key does not exist in the dictionary.